### PR TITLE
feat: squad page tutorial guides

### PR DESCRIPTION
**Pull request by capJavert**
### Describe what this PR does
- Add tutorial guides for posting and inviting members (also implements WT-1157) # done
- Add TutorialGuide component (it also has arrow logic for all 4 positions)

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-1156 #done

### CODE REVIEW

File: packages/shared/src/components/tutorial/TutorialGuide.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  arrowPlacement?: 'top' | 'right' | 'bottom' | 'left';
+  children: ReactNode;
+};
+
+const positionMapClassName = {
+  top: 'flex-col',
+  right: 'flex-row-reverse',
+  bottom: 'flex-col-reverse',
+};
+const arrowMapClassName = {
```
Comment (rebelchris) [inline packages/shared/src/components/tutorial/TutorialGuide.tsx:17]:
```suggestion
const positionToArrowClassName = {
```

File: packages/shared/src/components/tutorial/TutorialGuide.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import { IconSize } from '../Icon';
+import DrawnArrowIcon from '../icons/DrawnArrow';
+
+type TutorialGuideProps = {
+  className?: string;
+  arrowPlacement?: 'top' | 'right' | 'bottom' | 'left';
+  children: ReactNode;
+};
+
+const positionMapClassName = {
```
Comment (rebelchris) [inline packages/shared/src/components/tutorial/TutorialGuide.tsx:12]:
```suggestion
const positionToContainerClassName = {
```

File: packages/shared/src/components/squads/SquadPageHeader.tsx [hunk 3]
```diff
@@ -27,8 +31,19 @@ export function SquadPageHeader({
 }: SquadPageHeaderProps): ReactElement {
   const { sidebarRendered } = useSidebarRendered();
 
+  const sharePostTutorial = useTutorial({
+    key: TutorialKey.SHARE_SQUAD_POST,
+  });
+
+  const isDesktop = useMedia([laptop.replace('@media ', '')], [true], false);
```
Comment (rebelchris) [inline packages/shared/src/components/squads/SquadPageHeader.tsx:38]: I think we can use sidebarRendered for this rather?

File: packages/shared/src/components/squads/SquadPageHeader.tsx [hunk 4]
```diff
@@ -27,8 +31,19 @@ export function SquadPageHeader({
...
+
   return (
-    <FlexCol className="relative items-center laptop:items-start px-6 pb-20 tablet:pb-20 laptop:pb-14 mb-6 w-full tablet:border-b laptop:px-[4.5rem] min-h-20 border-theme-divider-tertiary">
+    <FlexCol
+      className={classNames(
+        'relative items-center laptop:items-start px-6 pb-20 tablet:pb-20 laptop:pb-14 mb-6 w-full tablet:border-b laptop:px-[4.5rem] min-h-20 border-theme-divider-tertiary',
+        sharePostTutorial.isActive && 'laptop:mb-28 mb-28',
```
Comment (rebelchris) [inline packages/shared/src/components/squads/SquadPageHeader.tsx:44]: I think the idea is that this always has the margin now, regardless of wether the post is shown.

File: packages/shared/src/components/tutorial/TutorialGuide.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        positionMapClassName[arrowPlacement],
+        className,
+      )}
+    >
+      <DrawnArrowIcon
+        className={arrowMapClassName[arrowPlacement]}
+        size={IconSize.Large}
+      />
+      {typeof children === 'string' ? (
+        <span className="typo-title2 text-theme-label-primary">{children}</span>
```
Comment (rebelchris) [inline packages/shared/src/components/tutorial/TutorialGuide.tsx:41]: Mobile it renders `title3` I see in the design now

File: packages/shared/src/components/squads/SquadHeaderBar.tsx [hunk 6]
```diff
@@ -36,19 +38,39 @@ export function SquadHeaderBar({
...
     >
-      <Button
-        className="btn-primary"
-        onClick={() => trackAndCopyLink()}
-        icon={<AddUserIcon />}
-        disabled={copying}
+      <div
+        className={classNames(
+          'relative',
+          copyLinkTutorial.isActive && 'laptop:m-0 mb-14 tablet:mb-10',
```
Comment (rebelchris) [inline packages/shared/src/components/squads/SquadHeaderBar.tsx:53]: Raised a question for product now about this shift.

**Review by rebelchris — CHANGES_REQUESTED** [review]
Really nice work Ante, looks pretty solid.
One generic issue: You forgot to implement it for the SquadTourPopup element, which is actually the main one that will show.

File: packages/shared/src/components/tutorial/TutorialGuide.tsx (hunk 5 above)
Comment (capJavert) [reply packages/shared/src/components/tutorial/TutorialGuide.tsx:41]: Oh, this must have been changed sometime yesterday, I'll update.

File: packages/shared/src/components/squads/SquadPageHeader.tsx (hunk 4 above)
Comment (rebelchris) [reply packages/shared/src/components/squads/SquadPageHeader.tsx:40]: Checked and we decided to keep as implemented

File: packages/shared/src/components/squads/SquadHeaderBar.tsx (hunk 6 above)
Comment (rebelchris) [reply packages/shared/src/components/squads/SquadHeaderBar.tsx:53]: Checked and we decided to keep as implemented

**Review by rebelchris — APPROVED** [review]
Ready to merge @capJavert 🥳

**Pull request merged**